#161966972 Add Functionality for User Registration

dmithamo1 opened the pull request:
#### What does this PR do?
This PR delivers the functionality that allows users to create accounts on `Author's Haven`.
#### Description of Task to be completed?
- Build out the `Registration Form Component`
- Configure the necessary `Actions and Reducers`
- Link the `Register` component with the rest of the `Author's Haven` platform
- Connect to backend and consume the `API`
- Write tests for the functionality

#### How should this be manually tested?
1. Execute the following commands in Terminal:
```
git clone https://github.com/andela/ah-jumanji-frontend.git
git checkout ft-user-registration-161966972
cd ah-jumanji-frontend
touch .env
```
2. Add the following to the `.env`:
```
baseUrl='http://127.0.0.1:8000'
registerEndpoint='/api/users/register'
activateUserUrl='/api/users/activate'
```
3. Continue in `Terminal`
```
source .env
npm install
npm start
``` 
4. The app auto-opens on your browser at `localhost:3000`
5. Visit the `Register` page and key in the required input then submit.
6. An email should be sent to the address provided upon successful registration, failing which an appropriate error message will be shown.
7. To run tests written for this, in `Terminal`, execute:
```
npm test
```

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
[#161966972](https://www.pivotaltracker.com/story/show/161966972)
#### Screenshots (if appropriate)
`1. Registration page - Errors`
![image](https://user-images.githubusercontent.com/45358371/51261415-3def8280-19c1-11e9-9f52-c220490c2633.png)

`2. On successful registration` 
![image](https://user-images.githubusercontent.com/45358371/51261483-67a8a980-19c1-11e9-8e5b-2417a0e2799e.png)

#### Questions:

#### Checklist:

- [x] My code follows the style guidelines of this project
- [ ] At least 2 people have reviewed my PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] My PR has one commit.

Commit:
feat(user registration): enable users to create accounts on the platform
- Add registration components
- Add reducer function
- Add tests
- Add notification alerts
[Starts #161966972]"

Commit:
feat(user registration): enable users to create accounts on the platform
- Add registration components
- Add reducer function
- Add tests
- Add notification alerts
[Finishes #161966972]"

Commit:
feat(user registration): enable users to create accounts on the platform
[Starts #161966972]

Commit:
feat(user registration): enable users to create accounts on the platform
- Add registration components
- Add reducer function
- Add tests
- Add notification alerts
[Starts #161966972]"

Commit:
feat(user registration): enable users to create accounts on the platform
- Add registration components
- Add reducer function
- Add tests
- Add notification alerts
[Starts #161966972]"

fabzer0 commented:
Raise your coverage

GransonO reviewed (commented):
I see you still have an issue with code climate linting.

mashafrancis committed:
feat(user registration): enable users to create accounts on the platform
- Add registration components
- Add reducer function
- Add tests
- Add notification alerts
[Starts #161966972]

mashafrancis committed:
feat(user registration): enable users to create accounts on the platform
- Add registration components
- Add reducer function
- Add tests
- Add notification alerts
[Finishes #161966972]
